### PR TITLE
Apply example_ynh (#37)

### LINKED ISSUE #32: Installation failed due to composer
### Describe the bug

The installation failed because of
```
2021-09-08 22:24:40,968: WARNING -   [RuntimeException]
2021-09-08 22:24:40,969: WARNING -   The HOME or COMPOSER_HOME environment variable must be set for composer to run correctly
```
I guess, but there is a [log](https://paste.yunohost.org/raw/fuzetubohi). 

### Context

- Hardware: some small box
- YunoHost version: 4.2.8.2
- I have access to my server: SSH and web interface 
- Using, or trying to install package version/branch: stabel movim, just via webinterface. /yunohost/admin... 

### Logs

[log](https://paste.yunohost.org/raw/fuzetubohi)

### LINKED ISSUE #35: cache directory permission incorrect
### Describe the bug
On fresh installation, in /var/www/movim/ cache, log and public/cache are owned by root and belong to group root.
They should belong to user movim and group www-data like rest of directory.

### Context

- Hardware: Kimsufi (remote server)
- YunoHost version: 4.2.8
- I have access to my server: Through SSH
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: no
 
### Steps to reproduce

On fresh install,
sudo yunohost app install movim
From browser, connect to movim instance or admin page.
Error message inviting to check syslog.
syslog reports permissions errors.

### Expected behavior

Movim usable

### Logs

```
Oct 02 03:54:14 domain.tld movim[29228]: movim.ERROR: fopen(/var/www/movim/cache/locales.ini.cache): failed to open stream: Permission denied in /var/www/movim/src/Movim/i18n/Locale.php 
Oct 02 03:54:14 domain.tld movim[29228]: PHP Warning:  fopen(/var/www/movim/cache/locales.ini.cache): failed to open stream: Permission denied in /var/www/movim/src/Movim/i18n/Locale.php

```
Changing ownership of these directories to movim:www-data allowed me to login and access admin page.

### PR DESCRIPTION
**Pull request by ericgaspar**
- *Fix #35*
- *Fix #32*
- *Fix error during package_check when `Validating that the app movim can/can't be accessed with its URL...` => `An error occured, check syslog for more information An error occured,`*

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

File: scripts/install [hunk 1]
```diff
@@ -54,148 +57,143 @@ ynh_app_setting_set --app=$app --key=admin --value=$admin
...
 # DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
-ynh_script_progression --message="Setting up source files..." --weight=4
+ynh_script_progression --message="Setting up source files..."
 
 ynh_app_setting_set --app=$app --key=final_path --value=$final_path
 # Download, check integrity, uncompress and patch the source from app.src
 ynh_setup_source --dest_dir="$final_path"
 
+ynh_replace_string --match_string="0.3.5" --replace_string="0.4.1" --target_file="$final_path/composer.json"
```
Comment (alexAubin) [inline scripts/install:107]: Ideally a comment wouldnt hurt to explain what this is about :stuck_out_tongue:

File: scripts/upgrade [hunk 2]
```diff
@@ -56,47 +81,26 @@ if ynh_legacy_permissions_exists; then
...
 
 if [ "$upgrade_type" == "UPGRADE_APP" ]
 then
-	ynh_script_progression --message="Upgrading source files..." --weight=1
+	ynh_script_progression --message="Upgrading source files..."
 
 	# Download, check integrity, uncompress and patch the source from app.src
 	ynh_setup_source --dest_dir="$final_path"
+
+	ynh_replace_string --match_string="0.3.5" --replace_string="0.4.1" --target_file="$final_path/composer.json"
```
Comment (alexAubin) [inline scripts/upgrade:104]: (same here)

**Review by alexAubin — APPROVED** [review]

File: scripts/install (hunk 1 above)
Comment (yalh76) [reply scripts/install:107]: Information added at https://github.com/YunoHost-Apps/movim_ynh/pull/38/commits/fb22b44a57b18d62a47cbbd7a47c96bc1fb2a2bb

File: scripts/upgrade (hunk 2 above)
Comment (yalh76) [reply scripts/upgrade:104]: (same here)

**Review by yalh76 — APPROVED** [review]

**Pull request merged**